apps/dashboard/services.py

```diff
@@ -423,6 +424,36 @@ def get_tag_analytics_data(self, **filters) -> dict[str, Any]:
         DashboardCache.set_cached_data(self.team, cache_key, data)
         return data
 
+    def get_average_response_time_data(self, granularity: str = "daily", **filters) -> list[dict[str, Any]]:
+        """Calculate average response time per period based on Trace table"""
+        cache_key = f"average_response_time_{granularity}_{hash(str(sorted(filters.items())))}"
+        cached_data = DashboardCache.get_cached_data(self.team, cache_key)
+        if cached_data:
+            return cached_data
+
+        querysets = self.get_filtered_queryset_base(**filters)
+        sessions = querysets["sessions"]
+
+        trunc_func = self._get_trunc_function(granularity)
+
+        avg_response_stats = (
+            Trace.objects.filter(session__in=sessions)
+            .annotate(period=trunc_func("session__created_at"))
```

Comment:
This should use the `Trace.created_at` (or whatever date on the trace object) as the anchor date.